EXPERIMENTS.md

```diff
@@ -44,3 +44,11 @@ Take `buildkite-agent artifact upload coverage\report.xml` as an example:
 - With this experiment enabled, it would be `s3://example/coverage/report.xml`.
 
 **Status**: a major improvement for Windows compatibility, we'd like this to be the standard behaviour in 4.0. 👍👍
+
+### `resolve-commit-after-checkout`
+
+When BUILDKITE_COMMIT is HEAD, this updates BUILDKITE_COMMIT to the resolved commit after checkout.
```

Comment:
Looks like the behaviour actually resolves any `BUILDKITE_COMMIT`, not just `HEAD`. That means it'll work for things like `HEAD^` and `HEAD~2` and I guess `master` and things like that. I think it'll also resolve commit hashes, presumably to themselves, but it would expand short commit hashes to full ones?

Do you think `resolveCommit()` should be a no-op if `BUILDKITE_COMMIT` already looks like a commit SHA? It might be tricky to know for sure. Or maybe it's harmless (and safer) always running it. 🤷‍♂️

Either way, I think this line of doc should say something like:

> Resolve `BUILDKITE_COMMIT` (which supports refs like `HEAD`) to a specific commit hash after checkout.

Reply:
I did some testing:

> Using resolve-commit-after-checkout experiment 🧪
> Updating BUILDKITE_COMMIT from "HEAD" to "5051b52eb78ca152edaeeed10743ed582f3d176a"

So far so good.

---

> Using resolve-commit-after-checkout experiment 🧪
> Updating BUILDKITE_COMMIT from "HEAD~1" to "d1be6bc8485bec8475f445a43ecd772efe3c14be"

I guess it's nice that it works on more interesting refs…

---

> Using resolve-commit-after-checkout experiment 🧪
> Updating BUILDKITE_COMMIT from "master" to "d1be6bc8485bec8475f445a43ecd772efe3c14be"

… including branches.

---

> Using resolve-commit-after-checkout experiment 🧪
> Updating BUILDKITE_COMMIT from "d1be6b" to "d1be6bc8485bec8475f445a43ecd772efe3c14be"

It does indeed expand short commit hashes, which is probably good?

---

> Using resolve-commit-after-checkout experiment 🧪
> Updating BUILDKITE_COMMIT from "d1be6bc8485bec8475f445a43ecd772efe3c14be" to "d1be6bc8485bec8475f445a43ecd772efe3c14be"

But that ends up looking weird when the `BUILDKITE_COMMIT` is already a commit SHA.

Reply:
> > Using resolve-commit-after-checkout experiment 🧪
> > Updating BUILDKITE_COMMIT from "d1be6bc8485bec8475f445a43ecd772efe3c14be" to "d1be6bc8485bec8475f445a43ecd772efe3c14be"
> 
> But that ends up looking weird when the BUILDKITE_COMMIT is already a commit SHA.

Fixed this one in ffb6ac48